File: adf_core_python/core/gateway/message/ma_exec_response.py
from rcrscore.config.config import Config
from rcrscore.messages import KAControlMessage
from rcrscore.proto import RCRSProto_pb2
from rcrscore.urn.control_message import ControlMessageURN

from adf_core_python.core.gateway.message.urn.urn import (
    ComponentModuleMSG,
    ModuleMSG,
)


class MAExecResponse(KAControlMessage):
    def __init__(self, message_proto: RCRSProto_pb2.MessageProto) -> None:
        self.result = Config()
        self.read(message_proto)

    def read(self, message_proto: RCRSProto_pb2.MessageProto) -> None:
        self.module_id: str = message_proto.components[
            ComponentModuleMSG.ModuleID
        ].stringValue
        result = message_proto.components[ComponentModuleMSG.Result].config
        for key, value in result.data.items():
            self.result.set_value(key, value)

    @staticmethod
    def get_urn() -> ControlMessageURN:
        return ModuleMSG.MA_EXEC_RESPONSE  # type: ignore

File: java/lib/src/main/java/adf_core_python/core/gateway/message/MAExecResponse.java
package adf_core_python.core.gateway.message;

import adf_core_python.core.gateway.message.urn.ModuleMessageComponentURN;
import adf_core_python.core.gateway.message.urn.ModuleMessageURN;
import rescuecore2.config.Config;
import rescuecore2.messages.AbstractMessage;
import rescuecore2.messages.components.ConfigComponent;
import rescuecore2.messages.components.StringComponent;
import rescuecore2.messages.protobuf.RCRSProto;

import java.io.IOException;
import java.io.InputStream;

public class MAExecResponse extends AbstractMessage {
    private final StringComponent moduleID;
    private final ConfigComponent result;

    public MAExecResponse(String moduleID, Config result) {
        this();
        this.moduleID.setValue(moduleID);
        this.result.setConfig(result);
    }

    public MAExecResponse(InputStream inputStream) throws IOException {
        this();
        read(inputStream);
    }

    public MAExecResponse(RCRSProto.MessageProto messageProto) {
        this();
        fromMessageProto(messageProto);
    }

    private MAExecResponse() {
        super(ModuleMessageURN.MA_EXEC_RESPONSE);
        this.moduleID = new StringComponent(ModuleMessageComponentURN.ModuleID);
        this.result = new ConfigComponent(ModuleMessageComponentURN.Result);
        addMessageComponent(moduleID);
        addMessageComponent(result);
    }

    public String getModuleID() {
        return this.moduleID.getValue();
    }

    public Config getResult() {
        return this.result.getConfig();
    }
}
